src/checkers/inference/solver/backend/maxsat/MaxSatSolver.java

```diff
@@ -76,7 +74,20 @@ public MaxSatBackEnd(Map<String, String> configuration, Collection<Slot> slots,
         this.convertAll();
         this.serializationEnd = System.currentTimeMillis();
 
-        generateOneHotClauses(hardClauses);
+        //TODO: Would it be better to have a type restriction on Translator class
+        // that a backend requires? E.g. for MaxSatBackend, the translator passed
+        // in has to be a sub-class of MaxSatTranslator.
+        //
+        // The pros would be we can have more flexible way of interactions between
+        // translators and backends without losing type safety.
+        //
```

Comment:
are there other solutions that can provide the same level of safety?

Reply:
This comment is outdate. The solution is actually reduce the type parameters on BackEnd from 2 to 1.

I will remove it.

Also, the type casting on line 87 makes no more sense, as `formatTranslator` is already an instance of `MaxSatFormatTranslator`.

Will clean this.